### PR TITLE
✨ Import grouping linter changes

### LINKED ISSUE #510: Linting for import groups
We already have some linting for imports [here](https://github.com/operator-framework/operator-controller/blob/2c3176d3308ebf4a9b9e2c6120029586ae4077f0/.golangci.yaml#L31-L50), but would be nice to have linting for import groups.

I'm thinking the the following grouping:

* Standard library
* 3rd-party packages excluding `github.com/operator-framework/*`
* 3rd-party package from `github.com/operator-framework/*`
* Local imports from `github.com/operator-framework/operator-controller/*`

I find this grouping helpful becuase it is very quick to distinguish OF dependencies from other 3rd party dependenices.

I think https://golangci-lint.run/usage/linters/#gci might be able to help with that.

### PR DESCRIPTION
**Pull request by trgeiger**
# Description
This PR enables the gci linter in golangci-lint and adds configuration to group imports as follows:
- standard library
- 3rd party packages
- packages from operator-framework org
- local imports from operator-framework/operator-controller

Closes #510 

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Review by m1kola — APPROVED** [review]
Looks great! Thanks for taking care of this.

Would you be able to do similar change in other OLMv1 repos (e.g. [catalogd](https://github.com/operator-framework/catalogd) and [rukpak](https://github.com/operator-framework/rukpak))?

File: .golangci.yaml [hunk 1]
```diff
@@ -24,6 +25,14 @@ linters:
   - whitespace
 
 linters-settings:
+  gci:
+    sections:
+      - standard
+      - default
+      - prefix(github.com/operator-framework)
+      - prefix(github.com/operator-framework/operator-controller)
```
[the remark this answers is not in the document]
Comment (trgeiger) [reply .golangci.yaml:33]: That's a good idea, I'll throw up another small PR soon.

[the remark this answers is not in the document]
Comment (trgeiger) [reply .golangci.yaml:33]: Would we want dot or alias groups?

[the remark this answers is not in the document]
Comment (m1kola) [reply .golangci.yaml:33]: I would say we do not need a separate group for those.

I would even consider disallowing dot imports.

[the remark this answers is not in the document]
Comment (trgeiger) [reply .golangci.yaml:33]: Looks like the localmodule section in gci hasn't hit golangci-lint yet

**Pull request merged**